torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -479,13 +486,20 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
       at::AutoNonVariableTypeMode non_variable_type_mode;
       // take the short-cut for launch if we see a recorded input set again
       launch_params = executor_entry->launch_params;
-      for (size_t i = 0; i < executor_entry->output_sizes.size(); i++) {
-        allocated_outputs.push_back(at::native::empty_cuda(
-            executor_entry->output_sizes[i],
-            executor_entry->output_types[i],
-            c10::nullopt,
-            options_.device,
-            c10::nullopt));
+      if (outputs.empty() || outputs.size() != fusion_.outputs().size()) {
```

Comment:
Doesn't the rhs of the `||` cover the lhs case?

Reply:
I'm confused why we set `allocated_outputs` driectly to `outputs` then have this second part of the conditional. Can you explain the logic here?

Reply:
Good catch. I should have removed the lhs of `||`.

In terms of the check, for some reason the function `runFusion` allows user to pass in output instead of allocating them inside this function. Those are used by some tests.
So there's no guarantee that provided output would actually work. Hence the check. There was a bug before this PR, that the provided outputs were ignored when we get a cached kernel.